lib/Random/RandomGenerator.cpp

```diff
@@ -100,7 +100,10 @@ int32_t RandomDevice::random(int32_t left, int32_t right) {
     return static_cast<int32_t>(random());
   }
 
-  uint32_t range = static_cast<uint32_t>(right - left + 1);
+  TRI_ASSERT(right > left);
+  int64_t r = static_cast<int64_t>(right) - static_cast<int64_t>(left) + 1;
```

Comment:
if the assert holds wouldn't  `static_cast<int64_t>(right - left)+1` be sufficient